### ISSUE TITLE
Option removeAdditional breaks custom keywords from 6.9.0

### ISSUE THREAD

**Issue by jandolezal71** [ajv-validator/ajv#955]
Hi,

we upgraded our AJV from `6.5.0` to `6.9.0` and we are struggling with the validation error:

```
data: null
dataPath: ".name"
keyword: "oneOf"
message: "should match exactly one schema in oneOf"
params: {passingSchemas: [0, 1]}
passingSchemas: [0, 1]
parentSchema: {oneOf: [{$ref: "trimmedString"}, {type: "null"}]}
oneOf: [{$ref: "trimmedString"}, {type: "null"}]
schema: [{$ref: "trimmedString"}, {type: "null"}]
0: {$ref: "trimmedString"}
1: {type: "null"}
```

Our schema is:

```
export default {
	type: 'object',
	properties: {
		name: {
			oneOf: [
				{ $ref: 'trimmedString' },
				{ type: 'null' },
			],
		},
	},
	required: [ 'name' ],
};
```

`trimmedString` is keyword:
```
{
	type: 'string',
	minTrimmedLength: 1,
};
```

and `minTrimmedLength` is keyword:

```
{
	compile (minTrimmedLength) {
		return function (sample) {
			sample = sample.trim();

			return sample.length >= minTrimmedLength;
		};
	},
}
```

Input is: 

```
{
  "name": null,
}
```

In version `6.5.0` everything works fine, but in `6.9.0` is that error. Whats wrong?

**Comment by epoberezkin** [comment]
Difficult to say... Would you be able to post a code sample to runkit.com? (it's easy to run there against different versions and try changing things)

**Comment by epoberezkin** [comment]
Alternatively you can try running your code against different Ajv versions to identify exactly which release caused the change.

**Comment by jandolezal71** [comment]
Sure, there you go (it behave exactly as i decribed)

https://runkit.com/jandolezal71/5c6fb3cd3c877f00138f1a40

**Comment by epoberezkin** [comment]
Yes, it's definitely the same issue as linked - if you remove removeAdditional it should work.

**Comment by epoberezkin** [comment]
Fixing it.

**Comment by epoberezkin** [comment]
as a recommendation, you may want to limit keyword to strings with "type" in definition, maybe you have it in your code.

**Comment by epoberezkin** [comment]
fixed in 6.9.2

**Comment by jandolezal71** [comment]
Thank you.
